staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/utils.go

```diff
@@ -203,31 +203,83 @@ func VerifyVolumePathsForVMDevices(vmDevices object.VirtualDeviceList, volPaths
 
 // isvCenterDeprecated takes vCenter version and vCenter API version as input and return true if vCenter is deprecated
 func isvCenterDeprecated(vCenterVersion string, vCenerAPIVersion string) (bool, error) {
-	minvcversion, err := semver.New(MinvCenterVersion)
-	vcdeprecated := false
+	var vcversion, vcapiversion, minvcversion VCVersion
+	var err error
+	err = vcversion.New(vCenterVersion)
 	if err != nil {
-		return false, fmt.Errorf("failed to get parse vCenter version: %s. err: %+v", MinvCenterVersion, err)
-	} else {
-		vcversion, err := semver.New(vCenterVersion)
+		return false, fmt.Errorf("failed to get parse vCenter version: %s. err: %+v", vCenterVersion, err)
+	}
+	err = vcapiversion.New(vCenerAPIVersion)
+	if err != nil {
+		return false, fmt.Errorf("failed to get parse vCenter API version: %s. err: %+v", vCenerAPIVersion, err)
+	}
+	err = minvcversion.New(MinvCenterVersion)
+	if err != nil {
+		return false, fmt.Errorf("failed to get minimum vCenter version: %s. err: %+v", MinvCenterVersion, err)
```

Comment:
nit: failed to parse minimum vCenter version ...

Reply:
fixed.